alicloud/resource_alicloud_ess_scalinggroup_vserver_groups.go

```diff
@@ -0,0 +1,264 @@
+package alicloud
+
+import (
+	"fmt"
+	"strconv"
+
+	"github.com/aliyun/alibaba-cloud-sdk-go/sdk/requests"
+	"github.com/aliyun/alibaba-cloud-sdk-go/services/ess"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/terraform-providers/terraform-provider-alicloud/alicloud/connectivity"
+)
+
+func resourceAlicloudEssScalingGroupVserverGroups() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAliyunEssVserverGroupsCreate,
+		Read:   resourceAliyunEssVserverGroupsRead,
+		Update: resourceAliyunEssVserverGroupsUpdate,
+		Delete: resourceAliyunEssVserverGroupsDelete,
+		Schema: map[string]*schema.Schema{
+			"scaling_group_id": {
+				Type:     schema.TypeString,
+				ForceNew: true,
+				Required: true,
+			},
+			"vserver_groups": {
+				Type:     schema.TypeSet,
+				Optional: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"loadbalancer_id": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+						"vserver_attributes": {
+							Type:     schema.TypeSet,
+							Required: true,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"vserver_group_id": {
+										Type:     schema.TypeString,
+										Required: true,
+									},
+									"port": {
+										Type:     schema.TypeInt,
+										Required: true,
+									},
+									"weight": {
+										Type:     schema.TypeInt,
+										Required: true,
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func resourceAliyunEssVserverGroupsCreate(d *schema.ResourceData, meta interface{}) error {
+	d.SetId(d.Get("scaling_group_id").(string))
+	return resourceAliyunEssVserverGroupsUpdate(d, meta)
+}
+
+func resourceAliyunEssVserverGroupsRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*connectivity.AliyunClient)
+	essService := EssService{client}
+	object, err := essService.DescribeEssScalingGroup(d.Id())
+	if err != nil {
+		if NotFoundError(err) {
+			d.SetId("")
+			return nil
+		}
+		return WrapError(err)
+	}
+	d.Set("scaling_group_id", object.ScalingGroupId)
+	d.Set("vserver_groups", object.VServerGroups.VServerGroup)
+
+	return nil
+}
+
+func resourceAliyunEssVserverGroupsUpdate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*connectivity.AliyunClient)
+	if d.HasChange("vserver_groups") {
```

Comment:
The update method have more than one API invoking and please set the partial for it.